### ISSUE TITLE
Runelite not automatically installed/detected

### ISSUE THREAD

**Issue by jullian-chavez** [TormStorm/jagex-launcher-linux#43]
After messing with this for hours, I finally got the Jagex launcher installed, launched, and logged in. However, the launcher does not detect either the Jagex client or Runelite. It just shows an "Install" button next to the client selector dropdown. I have Runelite installed separately, but I'm not sure what the launcher is expecting given that it's a windows application. Can it detect my linux install of Runelite? Did something go wrong in the installation process of the launcher? The documentation says that Runelite will automatically be installed and is able to be selected from the client dropdown, so i'm confused what is going on. Thanks for any help!

**Comment by nmlynch94** [comment]
Hello. The RuneLite launcher should be installed automatically as part of the process. I will test this later today to be sure it's still working. In the meantime, here are a couple things you can check:
1. Make sure the installation is done to an empty directory. Lutris doesn't block you from installing to a non-empty one, and it'll break the install. 
2. Lutris is supposed to default to the latest Proton-GE runner. There is a bug in the latest version causing it not to do that unless it is already installed. Make sure in your runners tab, you check the versions next to the Wine section and install the latest Proton-GE version. Once that is done, try re-running the installer in a fresh directory.

**Comment by nmlynch94** [comment]
For what it's worth, I tested on a fresh system and confirmed RuneLite is still installing properly on my end. Please try the above and, if it doesn't work, we'll likely need your OS information and some logs.

**Comment by Yannjoel** [comment]
I'm having the same problem on Ubuntu 22.04.2:
As far as I can tell, the runelite files are correctly installed at "~/Games/jagex-launcher/drive_c/Program Files (x86)/Jagex Launcher/Games/RuneLite"

The file and their permissions in the directory are :
```
drwxrwxr-x 2 myuser myuser     4096 Jul 19 13:04 .
drwxrwxr-x 4 myuser myuser     4096 Jul 19 13:04 ..
-rwxrw-r-- 1 myuser myuser 46253248 Jul 19 13:04 RuneLite.AppImage
lrwxrwxrwx 1 myuser myuser      105 Jul 19 13:04 RuneLite.exe -> '/home/myuser/Games/jagex-launcher/drive_c/Program Files (x86)/Jagex Launcher/Games/RuneLite/runelite.sh'
-rwxrw-r-- 1 myuser myuser       79 Jul 19 13:04 runelite.sh
```

Nevertheless, the RuneLite-Launcher isn't detected inside the Jagex launcher. The install button just links to runelite.net:
![grafik](https://github.com/TormStorm/jagex-launcher-linux/assets/18733146/7b1e57ae-8179-46ce-8285-4c43c3f11b96)


Installing the normal Launcher from within the jagex Launcher installs it to "C:\Program Files (x86)\Jagex Launcher\Games\Old School RuneScape" - The game successfully launches with the official launcher

**Comment by Yannjoel** [comment]
A workaround, that solved it for me, was to install the windows RunLite-Launcher with
`env WINEPREFIX=/home/myuser/Games/jagex-launcher /home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine RuneLiteSetup.exe`

This installed the windows runlite launcher to "/home/myuser/Games/jagex-launcher/drive_c/users/myuser/AppData/Local"

I then replaced the windows RuneLite.exe with the simlink to "/home/myuser/Games/jagex-launcher/drive_c/Program Files (x86)/Jagex Launcher/Games/RuneLite/runelite.sh"

Now the jagex-launcher shows me the option to start runelite and launches the linux version.

**Comment by nmlynch94** [comment]
Your sleuthing is correct. The lutris script is supposed to have set that registry path for you to the existing installation. Thank you for sharing your workaround.

 We are getting several reports of this and aren't sure why it's failing in some cases. If anyone is able to reproduce by running `lutris -i <.yml file>` or `flatpak run net.lutris.Lutris -i <.yml file>` for flatpak and share the logs, we may be able to find something in there.

**Comment by nmlynch94** [comment]
In the same vein as @Yannjoel 's solution, you can try editing the registry value manually which should be more straightforward in theory. Hightlight the jagex launcher then select Wine registry in the window shown
![image](https://github.com/TormStorm/jagex-launcher-linux/assets/40608755/a8627541-6e98-4473-858d-9c7d2630f56e)

Then, make sure this registry value exists with the following value:
![image](https://github.com/TormStorm/jagex-launcher-linux/assets/40608755/76239944-8a34-4543-b5ff-835d33436ddc)

**Comment by Yannjoel** [comment]
> 

I was able to reproduce it: Here is the snippet of the installation log:
```
DEBUG    2023-07-19 18:56:03,337 [command.on_stop:195]:Process 157407 has terminated with code 0
DEBUG    2023-07-19 18:56:04,194 [commands._monitor_task:444]:Return code: 0
DEBUG    2023-07-19 18:56:04,194 [interpreter._iter_commands:361]:Installer command: {'task': {'name': 'set_regedit', 'path': 'HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\RuneLite Launcher_is1', 'key': 'InstallLocation', 'value': 'C:\\\\Program Files (x86)\\\\Jagex Launcher\\\\Games\\\\RuneLite'}}
DEBUG    2023-07-19 18:56:04,195 [wine.set_regedit:48]:Setting [HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\RuneLite Launcher_is1]:InstallLocation="C:\\Program Files (x86)\\Jagex Launcher\\Games\\RuneLite"
DEBUG    2023-07-19 18:56:04,216 [wine.use_lutris_runtime:123]:/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine64 is provided by Lutris, using runtime
DEBUG    2023-07-19 18:56:04,287 [system.execute:74]:Executing /home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine64 regedit /S /home/myuser/.cache/lutris/winekeys.reg
DEBUG    2023-07-19 18:56:04,287 [system.execute:80]:__GL_SHADER_DISK_CACHE=1 __GL_SHADER_DISK_CACHE_PATH=/home/myuser/.cache/lutris/shaders/wine PULSE_LATENCY_MSEC=60 LD_LIBRARY_PATH=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib:/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib64:/lib/x86_64-linux-gnu:/lib/i386-linux-gnu:/lib:/lib32:/lib/i386-linux-gnu/sse2:/usr/lib/x86_64-linux-gnu/libfakeroot:/lib64:/lib64:/usr/lib:/usr/lib64:/usr/lib32:/usr/lib64:/usr/lib/i386-linux-gnu:/usr/lib/x86_64-linux-gnu:/home/myuser/.local/share/lutris/runtime/Ubuntu-18.04-i686:/home/myuser/.local/share/lutris/runtime/steam/i386/lib/i386-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/i386/lib:/home/myuser/.local/share/lutris/runtime/steam/i386/usr/lib/i386-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/i386/usr/lib:/home/myuser/.local/share/lutris/runtime/Ubuntu-18.04-x86_64:/home/myuser/.local/share/lutris/runtime/steam/amd64/lib/x86_64-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/amd64/lib:/home/myuser/.local/share/lutris/runtime/steam/amd64/usr/lib/x86_64-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/amd64/usr/lib WINEDEBUG=-all DXVK_LOG_LEVEL=none WINEARCH=win64 WINE=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine64 WINE_MONO_CACHE_DIR=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/mono WINE_GECKO_CACHE_DIR=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/gecko GST_PLUGIN_SYSTEM_PATH_1_0=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib64/gstreamer-1.0/:/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib/gstreamer-1.0/ WINEPREFIX=/home/myuser/Games/jagex-launcher WINEESYNC=0 WINEFSYNC=0 WINE_FULLSCREEN_FSR=1 DXVK_NVAPIHACK=0 DXVK_ENABLE_NVAPI=1 PROTON_BATTLEYE_RUNTIME=/home/myuser/.local/share/lutris/runtime/battleye_runtime PROTON_EAC_RUNTIME=/home/myuser/.local/share/lutris/runtime/eac_runtime WINEDLLOVERRIDES=d3d10core,d3d11,d3d12,d3d12core,d3d9,d3dcompiler_33,d3dcompiler_34,d3dcompiler_35,d3dcompiler_36,d3dcompiler_37,d3dcompiler_38,d3dcompiler_39,d3dcompiler_40,d3dcompiler_41,d3dcompiler_42,d3dcompiler_43,d3dcompiler_46,d3dcompiler_47,d3dx10,d3dx10_33,d3dx10_34,d3dx10_35,d3dx10_36,d3dx10_37,d3dx10_38,d3dx10_39,d3dx10_40,d3dx10_41,d3dx10_42,d3dx10_43,d3dx11_42,d3dx11_43,d3dx9_24,d3dx9_25,d3dx9_26,d3dx9_27,d3dx9_28,d3dx9_29,d3dx9_30,d3dx9_31,d3dx9_32,d3dx9_33,d3dx9_34,d3dx9_35,d3dx9_36,d3dx9_37,d3dx9_38,d3dx9_39,d3dx9_40,d3dx9_41,d3dx9_42,d3dx9_43,dxgi,nvapi,nvapi64,nvml=n;winemenubuilder= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/myuser/.steam/debian-installation/ STEAM_COMPAT_DATA_PATH=/home/myuser/Games/jagex-launcher STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId=lutris-game
DEBUG    2023-07-19 18:56:04,857 [interpreter._iter_commands:361]:Installer command: {'task': {'name': 'set_regedit', 'path': 'HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\RuneScape Launcher_is1', 'key': 'InstallLocation', 'value': 'C:\\\\Program Files (x86)\\\\Jagex Launcher\\\\Games\\\\RuneScape'}}
DEBUG    2023-07-19 18:56:04,858 [wine.set_regedit:48]:Setting [HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\RuneScape Launcher_is1]:InstallLocation="C:\\Program Files (x86)\\Jagex Launcher\\Games\\RuneScape"
DEBUG    2023-07-19 18:56:04,879 [wine.use_lutris_runtime:123]:/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine64 is provided by Lutris, using runtime
DEBUG    2023-07-19 18:56:04,941 [system.execute:74]:Executing /home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine64 regedit /S /home/myuser/.cache/lutris/winekeys.reg
DEBUG    2023-07-19 18:56:04,942 [system.execute:80]:__GL_SHADER_DISK_CACHE=1 __GL_SHADER_DISK_CACHE_PATH=/home/myuser/.cache/lutris/shaders/wine PULSE_LATENCY_MSEC=60 LD_LIBRARY_PATH=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib:/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib64:/lib/x86_64-linux-gnu:/lib/i386-linux-gnu:/lib:/lib32:/lib/i386-linux-gnu/sse2:/usr/lib/x86_64-linux-gnu/libfakeroot:/lib64:/lib64:/usr/lib:/usr/lib64:/usr/lib32:/usr/lib64:/usr/lib/i386-linux-gnu:/usr/lib/x86_64-linux-gnu:/home/myuser/.local/share/lutris/runtime/Ubuntu-18.04-i686:/home/myuser/.local/share/lutris/runtime/steam/i386/lib/i386-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/i386/lib:/home/myuser/.local/share/lutris/runtime/steam/i386/usr/lib/i386-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/i386/usr/lib:/home/myuser/.local/share/lutris/runtime/Ubuntu-18.04-x86_64:/home/myuser/.local/share/lutris/runtime/steam/amd64/lib/x86_64-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/amd64/lib:/home/myuser/.local/share/lutris/runtime/steam/amd64/usr/lib/x86_64-linux-gnu:/home/myuser/.local/share/lutris/runtime/steam/amd64/usr/lib WINEDEBUG=-all DXVK_LOG_LEVEL=none WINEARCH=win64 WINE=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine64 WINE_MONO_CACHE_DIR=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/mono WINE_GECKO_CACHE_DIR=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/gecko GST_PLUGIN_SYSTEM_PATH_1_0=/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib64/gstreamer-1.0/:/home/myuser/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/lib/gstreamer-1.0/ WINEPREFIX=/home/myuser/Games/jagex-launcher WINEESYNC=0 WINEFSYNC=0 WINE_FULLSCREEN_FSR=1 DXVK_NVAPIHACK=0 DXVK_ENABLE_NVAPI=1 PROTON_BATTLEYE_RUNTIME=/home/myuser/.local/share/lutris/runtime/battleye_runtime PROTON_EAC_RUNTIME=/home/myuser/.local/share/lutris/runtime/eac_runtime WINEDLLOVERRIDES=d3d10core,d3d11,d3d12,d3d12core,d3d9,d3dcompiler_33,d3dcompiler_34,d3dcompiler_35,d3dcompiler_36,d3dcompiler_37,d3dcompiler_38,d3dcompiler_39,d3dcompiler_40,d3dcompiler_41,d3dcompiler_42,d3dcompiler_43,d3dcompiler_46,d3dcompiler_47,d3dx10,d3dx10_33,d3dx10_34,d3dx10_35,d3dx10_36,d3dx10_37,d3dx10_38,d3dx10_39,d3dx10_40,d3dx10_41,d3dx10_42,d3dx10_43,d3dx11_42,d3dx11_43,d3dx9_24,d3dx9_25,d3dx9_26,d3dx9_27,d3dx9_28,d3dx9_29,d3dx9_30,d3dx9_31,d3dx9_32,d3dx9_33,d3dx9_34,d3dx9_35,d3dx9_36,d3dx9_37,d3dx9_38,d3dx9_39,d3dx9_40,d3dx9_41,d3dx9_42,d3dx9_43,dxgi,nvapi,nvapi64,nvml=n;winemenubuilder= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/myuser/.steam/debian-installation/ STEAM_COMPAT_DATA_PATH=/home/myuser/Games/jagex-launcher STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId=lutris-game
DEBUG    2023-07-19 18:56:05,495 [interpreter._iter_commands:379]:Commands 15 out of 15 completed
DEBUG    2023-07-19 18:56:05,495 [config.write_game_config:22]:Writing game config to jagex-launcher-installer-1689785765
DEBUG    2023-07-19 18:56:05,511 [http._request:88]:GET https://lutris.net/api/games/jagex-launcher
DEBUG    2023-07-19 18:56:05,963 [game.save:423]:Saving Jagex Launcher (wine) with config ID jagex-launcher-installer-1689785765
DEBUG    2023-07-19 18:56:05,963 [config.save:229]:Saving LutrisConfig(level=game, game_config_id=jagex-launcher-installer-1689785765, runner=wine) config to /home/myuser/.config/lutris/games/jagex-launcher-installer-1689785765.yml
DEBUG    2023-07-19 18:56:06,011 [lutris.add_to_path_cache:158]:Adding Jagex Launcher (wine) to path cache
DEBUG    2023-07-19 18:56:09,868 [system.remove_folder:321]:Removing folder /home/myuser/.cache/lutris/installer/jagex-launcher
DEBUG    2023-07-19 18:56:09,876 [application.on_app_window_destroyed:357]:Removed window InstallerWindowjagex-launcher
```

And just in case also the full installation log: 
[lutris-debug-log-install-jagexlauncher-ubuntu.txt](https://github.com/TormStorm/jagex-launcher-linux/files/12098292/lutris-debug-log-install-jagexlauncher-ubuntu.txt)

After the installation, the key is missing in the registry: 
![grafik](https://github.com/TormStorm/jagex-launcher-linux/assets/18733146/c5ba657e-bba9-4cd4-a16a-a68a1ec7259d)


Could it be, that it is installing the key to the wrong wine prefix? 
Early in the log it mentions another prefix location:
```
INFO     2023-07-19 18:54:27,936 [wine.create_prefix:98]:Creating a win64 prefix in /home/myuser/Games/jagex-launcher
```

**Comment by nmlynch94** [comment]
Thank you! I had a hunch it had something to do with placing the reg key in the wrong prefix, so that's a promising lead.

I have a working installation log on my home PC I can compare this to once I'm home later. Hopefully that'll shed some light on what about our setups are different that would cause this. 

If you are able, could you please try this version of the script and see if it fixes it? https://github.com/nmlynch94/jagex-launcher-linux/blob/patch-13/resources/jagexlauncher.yml. I added the prefix explicitly on those steps. Still unsure why it would use the correct prefix on some machines but not others.

**Comment by Yannjoel** [comment]
It worked for me, after changing the order of the tasks - I'm not quite sure why this has an effect.    
I found nothing, that explained why the order of the task should have a different outcome on some systems.

The following installation skript worked for me:
[jagexlauncher-patch13_modified.yml.txt](https://github.com/TormStorm/jagex-launcher-linux/files/12099000/jagexlauncher-patch13_modified.yml.txt)

**Comment by nmlynch94** [comment]
That's very strange. Appreciate you doing that testing. @jullian-chavez would you be willing to try the above script to see if it works for you as well?

**Comment by jullian-chavez** [comment]
@nmlynch94 That worked perfectly! Sorry I couldnt contribute more. I use linux as my main work OS every day, but I'm not a linux expert by any means. Took me forever to just get wine working and get this installed. I appreciate the help! I'm glad others were able to reproduce. I couldnt tell if I was doing something wrong. Thanks again!

**Comment by nmlynch94** [comment]
No need to apologize, thank you for reporting the issue. I'll make a PR to make the fixes above once I confirm it still works in my test boxes.

**Comment by TormStorm** [comment]
Fixed with #50
